src/libsync/propagateuploadng.cpp

```diff
@@ -387,6 +389,31 @@ void PropagateUploadFileNG::slotPutFinished()
         return;
     }
 
+    // Adjust the chunk size for the time taken.
+    //
+    // Dynamic chunk sizing is enabled if the server configured a
+    // target duration for each chunk upload.
+    double targetDuration = _propagator->account()->capabilities().desiredChunkUploadDuration();
+    if (targetDuration > 0) {
+        double uploadTime = job->msSinceStart();
+
+        auto correctedSize = static_cast<quint64>(
+                _currentChunkSize / uploadTime * targetDuration);
+
+        // There can be multiple chunk uploads going on at the same time.
+        // So don't force the chunk size to the new predicted best size
+        // and instead move it there gradually.
+        _propagator->_chunkSize = qBound(
+                _propagator->minChunkSize(),
+                (_propagator->_chunkSize + correctedSize) / 2,
```

Comment:
Both work. My reasoning was this:

The whole thing is heuristic. It's possibly that the code is executed after a single chunk finishes uploading when nothing else was going on, but it could also be that several chunks were competing for bandwidth when we get here. If we wanted to target a specific duration per chunk upload this would not give us the right result even if network throughput was completely constant.

Thus I expect `correctedSize` to over and undershoot a good size all the time. Using an exponential moving average like is a cheap way of smoothing chunk sizes a bit.

I could put this reasoning into the code or just remove the averaging, both are fine with me.